app/lib/efile/nj/nj1040_calculator.rb

```diff
@@ -325,11 +325,38 @@ def calculate_line_39
       end
 
       def is_ineligible_or_unsupported_for_property_tax_credit
-        StateFile::NjHomeownerEligibilityHelper.determine_eligibility(@intake) == StateFile::NjHomeownerEligibilityHelper::INELIGIBLE ||
-          Efile::Nj::NjPropertyTaxEligibility.ineligible?(@intake)
+        return true if Efile::Nj::NjPropertyTaxEligibility.ineligible?(@intake)
+
+        case @intake.household_rent_own
+        when "own"
+          StateFile::NjHomeownerEligibilityHelper.determine_eligibility(@intake) == StateFile::NjHomeownerEligibilityHelper::INELIGIBLE
+        when "rent"
+          StateFile::NjTenantEligibilityHelper.determine_eligibility(@intake) == StateFile::NjTenantEligibilityHelper::INELIGIBLE
+        when "both"
+          StateFile::NjTenantEligibilityHelper.determine_eligibility(@intake) == StateFile::NjTenantEligibilityHelper::INELIGIBLE &&
+            StateFile::NjHomeownerEligibilityHelper.determine_eligibility(@intake) == StateFile::NjHomeownerEligibilityHelper::INELIGIBLE
+        else
+          nil
+        end
       end
 
       def calculate_line_40a
+        if @intake.household_rent_own_both?
+          return nil unless @intake.rent_paid&.positive? || @intake.property_tax_paid&.positive?
+          property_tax_paid = @intake.property_tax_paid&.positive? ? @intake.property_tax_paid : 0
+          rent_paid = @intake.rent_paid&.positive? ? @intake.rent_paid * RENT_CONVERSION : 0
```

Comment:
[dust] - two above lines could be

```
property_tax_paid = [@intake.property_tax_paid, 0].max
rent_paid = [@intake.rent_paid * RENT_CONVERSION, 0].max
```

Reply:
Hm, this doesn't actually work because of the case where `@intake.property_tax_paid` or `@intake.rent_paid` could be nil. Either the integer comparison fails or the `*` operation fails on nil. It feels like I'd still need a ternary to test for `present` in this case, so I might as well just do the ternary as is? Any ideas on how to use the max syntax if possibly nil?

Reply:
i missed that one could be nil while the other is positive! you'd have to add `.to_i` like this then (`nil.to_i == 0`) 

```
property_tax_paid = [@intake.property_tax_paid.to_i, 0].max
rent_paid = [@intake.rent_paid.to_i * RENT_CONVERSION, 0].max
```

Reply:
Thanks so much! Updated it to use `to_f` so that our rounding logic still works, but this is great, thanks